packages/flutter_tools/test/general.shard/windows_version_validator_test.dart

```diff
@@ -169,4 +171,75 @@ OS 版本:          10.0.22621 暂缺 Build 22621
     expect(result.messages.length, 1, reason: 'The ValidationResult should have precisely 1 message');
     expect(result.messages[0].message, getProcessFailed.messages[0].message, reason: 'The ValidationMessage message should be the same');
   });
+
+  testWithoutContext('Falls back to pwsh when powershell is not on the path', () async {
+    final ProcessLister processLister = ProcessLister(
+      FakeProcessManager.list(
+        <FakeCommand>[
+          const FakeCommand(
+            command: <String>[
+              'powershell',
+              '-command',
+              'Get-Process | Format-List Path',
+            ],
+            exception: ProcessPackageExecutableNotFoundException('powershell'),
+            stdout: 'powershell',
+          ),
+          const FakeCommand(
+            command: <String>[
+              'pwsh',
+              '-command',
+              'Get-Process | Format-List Path',
+            ],
+            stdout: 'pwsh',
+          ),
+        ],
+      ),
+    );
+
+    try {
+      final ProcessResult result = await processLister.getProcessesWithPath();
+      expect(result.stdout, 'pwsh');
+    // ignore: avoid_catches_without_on_clauses
+    } catch (e) {
+      fail('Unexpected exception: $e');
+    }
+  });
+
+  testWithoutContext('getProcessesWithPath throws if both powershell and pwsh are not on PATH', () async {
+    final ProcessLister processLister = ProcessLister(
+      FakeProcessManager.list(
+        <FakeCommand>[
+          const FakeCommand(
+            command: <String>[
+              'powershell',
+              '-command',
+              'Get-Process | Format-List Path',
+            ],
+            exception: ProcessPackageExecutableNotFoundException('powershell'),
+            stdout: 'powershell',
+          ),
+          const FakeCommand(
+            command: <String>[
+              'pwsh',
+              '-command',
+              'Get-Process | Format-List Path',
+            ],
+            exception: ProcessPackageExecutableNotFoundException('pwsh'),
+            stdout: 'pwsh',
+          ),
+        ],
+      ),
+    );
+
+    try {
+      final ProcessResult result = await processLister.getProcessesWithPath();
+      fail('Should have thrown, but successfully ran ${result.stdout}');
+    } on ProcessPackageExecutableNotFoundException {
+      // Expected
+    // ignore: avoid_catches_without_on_clauses
```

Comment:
Is this line necessary? `flutter analyze` passes without it.

Reply:
Then again, the `Linux analyze` check is failing; so me running `flutter analyze` locally is apparently not sufficient 🤪